aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -19,11 +19,130 @@
 #include <ATen/cuda/detail/KernelUtils.h>
 #include <ATen/native/cuda/Loops.cuh>
 #include <ATen/native/cuda/DistributionTemplates.h>
+#include <ATen/native/Resize.h>
 #include <c10/cuda/CUDAMathCompat.h>
 
 namespace at {
 namespace native {
 
+// -----------------------------------
+// glu forward
+// -----------------------------------
+void glu_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "glu_cuda", [&]() {
+    using acc_t = at::acc_type<scalar_t, true>;
+    gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a_, scalar_t b_) -> scalar_t {
+      const acc_t a = a_;
+      const acc_t b = b_;
+      const acc_t one = acc_t(1);
+      const acc_t sigmoid = one / (one + std::exp(-b));
+      return a * sigmoid;
+    });
+  });
+}
+
+// -----------------------------------
+// glu backward
+// -----------------------------------
+
+// Byte offsets don't require multiplication by sizeof(T), so are slightly cheaper.
+// For fixed offsets, this removes all penalty from 64-bit indexing.
+template <typename T>
+__device__ T* byte_offset(T* ptr, int64_t offset) {
+  using byte_ptr_t = typename std::conditional<
+    std::is_const<T>::value, const char*, char*>::type;
+  return reinterpret_cast<T*>(
+    reinterpret_cast<byte_ptr_t>(ptr) + offset
+  );
+}
+
+template <typename scalar_t, typename OffsetCalc>
+__global__ void glu_backward_kernel(
+    int numel, scalar_t* gI, const scalar_t* I, const scalar_t* gO,
+    OffsetCalc offset_calculator,
+    int64_t gI_byte_offset, int64_t I_byte_offset) {
+  using acc_t = at::acc_type<scalar_t, true>;
+  // We explicitly iterate over the first half of the input tensor, and
+  // gI_byte_offset and I_byte_offset are the offsets to access the
+  // corresponding index in the second half of the tensor.
+  CUDA_KERNEL_LOOP(i, numel) {
+    const auto offsets = offset_calculator.get(i);
+    const acc_t a = I[offsets[1]];
+    const acc_t b = *byte_offset(I + offsets[1], I_byte_offset);
+    const acc_t gO_val = gO[offsets[2]];
+
+    const auto one = acc_t(1);
+    const acc_t sigmoid = one / (one + std::exp(-b));
+
+    auto* gA = gI + offsets[0];
+    *gA = sigmoid * gO_val;
+
+    auto* gB = byte_offset(gA, gI_byte_offset);
+    *gB = (one - sigmoid) * sigmoid * gO_val * a;
+  }
+}
+
+void launch_glu_backward_kernel(const TensorIteratorBase& iter,
+                                int64_t gI_stride, int64_t I_stride) {
+  const auto N = iter.numel();
+  auto offset_calculator = make_element_offset_calculator<3>(iter);
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(N > 0 && N <= std::numeric_limits<int32_t>::max());
+  int64_t grid = (N + NUM_THREADS - 1) / NUM_THREADS;
```

Comment:
NUM_THREADS and num_threads is 64, which is fine for vectorized TI kernels, but here I verified using num_threads=256 speeds things up for smaller sizes by ~10-15%.